### PR TITLE
Preliminar migration of itests into new inventory

### PR DESCRIPTION
**Pull request by lucasponce**
This is a WIP, marked as "Do Not Merge Yet".

I have moved the itests to make them compatible with the new inventory API and my next step is to execute them and validates that everything works.

But as I will need to rebase other works I have triggered the PR here to simplify potential problems.

### CODE REVIEW

File: hawkular-agent-itest-util/src/main/java/org/hawkular/agent/itest/util/ITestHelper.java [hunk 1]
```diff
@@ -274,14 +290,14 @@ private String getWithRetries(Request request) throws Throwable {
...
             long sleep, int attempts)
             throws Throwable {
         for (int i = 0; i < attempts; i++) {
-            Optional<Map.Entry<CanonicalPath, Blueprint>> resource = getBlueprintsByType(feed, rType, 0)
-                    .entrySet().stream()
+            Optional<ResourceWithType> resource = getResourceByType(feed, rType, 0)
+                    .stream()
                     .filter(e -> containing == null
-                            || ((Entity.Blueprint) (e.getValue())).getId().contains(containing))
+                            || e.getId().contains(containing)) // TODO [lponce] not sure if containing matches against id or name ?
```
Comment (jotak) [inline hawkular-agent-itest-util/src/main/java/org/hawkular/agent/itest/util/ITestHelper.java]: that would be name I think

File: hawkular-javaagent-itest-parent/hawkular-javaagent-all-itests/hawkular-javaagent-cmdgw-itest/src/test/java/org/hawkular/agent/ws/test/StandaloneDeployApplicationITest.java [hunk 2]
```diff
@@ -237,40 +247,49 @@ public void testLocalAndRemoteJMXOperations() throws Throwable {
...
-        invokeJMXOperations(localPath);
-
-        CanonicalPath remotePath = testHelper.feedPath(hawkularFeedId).resource("Remote JMX~org.hawkular.agent.itest:type=simple").get();
-        resource = testHelper.getBlueprintFromCP(remotePath);
-        Assert.assertTrue(resource.isPresent());
-
-        invokeJMXOperations(remotePath);
+        Collection<ResourceWithType> mbeans = testHelper.getResourceByType(hawkularFeedId, "Simple ITest MBean", 2);
+        Optional<ResourceWithType> localMbean = mbeans.stream()
+                // TODO [lponce] this is not 100% right, id should be opaque but Local JMX and Remote JMX have same name
```
Comment (jotak) [inline hawkular-javaagent-itest-parent/hawkular-javaagent-all-itests/hawkular-javaagent-cmdgw-itest/src/test/java/org/hawkular/agent/ws/test/StandaloneDeployApplicationITest.java:252]: other way, if you want to be less dependent on ID generation, would be to check `.contains("Local JMX")` maybe? (if it doesn't return other positives)

File: hawkular-javaagent/src/main/java/org/hawkular/agent/javaagent/cmd/UpdateCollectionIntervalsCommand.java [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+    private String[] parseMetricTypeId(String metricTypeId) {
+        String[] names = metricTypeId.split("~");
+        if (names.length != 2) {
+            throw new IllegalArgumentException(
+                    "MetricTypeId must be of form MetricTypeSetName~MetricTypeName: " + metricTypeId);
+        }
+        return names;
+    }
+
+    private String[] parseAvailTypeId(String availTypeId) {
```
Comment (jotak) [inline hawkular-javaagent/src/main/java/org/hawkular/agent/javaagent/cmd/UpdateCollectionIntervalsCommand.java:190]: So it's same as parseMetricTypeId, is it kept separate on purpose? (maybe more validation will be added later?)

File: hawkular-javaagent-itest-parent/hawkular-javaagent-all-itests/hawkular-javaagent-cmdgw-itest/src/test/java/org/hawkular/agent/ws/test/StandaloneDeployApplicationITest.java (hunk 2 above)
Comment (lucasponce) [reply hawkular-javaagent-itest-parent/hawkular-javaagent-all-itests/hawkular-javaagent-cmdgw-itest/src/test/java/org/hawkular/agent/ws/test/StandaloneDeployApplicationITest.java:252]: Yes, that would be great, but issue is the agent stores for MBeans the same name, on this case it hasn't info about Local JMX or Remote JMX.
I didn't want to dig here so I commented.
In general, I found that perhaps some names could be changed for more easy browsing on this new model.
Anyway, I will try to collect more feedback when I have all these tests up and running, but thx for the comment.
//cc @jmazzitelli

**Comment by lucasponce** [comment]
@jmazzitelli,
this PR is progressing and it covers mostly of the cmdgw itests.
Some things are not yet ready:
- itests related to configuration properties (commented in the PR waiting to resolve some doubts).
- itests related to domains (perhaps I need to check with you what is the final scenario for domain servers).
Tomorrow we can chat about it.

File: hawkular-javaagent/src/main/java/org/hawkular/agent/javaagent/cmd/UpdateCollectionIntervalsCommand.java (hunk 3 above)
Comment (lucasponce) [reply hawkular-javaagent/src/main/java/org/hawkular/agent/javaagent/cmd/UpdateCollectionIntervalsCommand.java:190]: Yes, perhaps this should be revisited it, I just brought back this class but definitely not all tests are passing yet.

**Comment by jotak** [comment]
We can discuss it later today, but perhaps this PR could be merged even if not all tests are fixed yet

**Comment by lucasponce** [comment]
I have rebased @jotak on this PR, now it is sync-ed.

As commented the status is the following:

```
[INFO] Hawkular Agent: Command Gateway Integration Tests .. SUCCESS [03:02 min]
[INFO] Hawkular Agent: WildFly Domain Environment Integration Tests FAILURE [01:22 min]
```

Most of the itests of CmdGw are passing but missing related to configuration properties and some of domains.

So, I am ok with whatever option, to wait until I can fix those ones, or consolidate this to not create a big PR.

Let me know what do you think.

**Comment by jotak** [comment]
I'm +1 for merging so that everybody can benefit from the work done so far

**Comment by jmazzitelli** [comment]
I'm merging this. I have a fix for the rest of the failures. I'll commit those soon.

**Pull request merged**